lib/elixir/lib/stream.ex

```diff
@@ -1420,7 +1420,6 @@ defmodule Stream do
       # Optimize the most common cases
       case next_fun.(next_acc) do
         {[], next_acc} -> {:opt, {:cont, acc}, next_acc}
```

Comment:
I'll move this downwards to the `else` branch as well to simplify things a bit. Also I've noticed that there's probably no test for this case when empty list is returned, so I'll add one ..tomorrow.

Reply:
Done.